### DISCUSSION TITLE
Using ldtk for maps instead

### DISCUSSION THREAD

**Discussion by MTRNord** [thecodingmachine/workadventure#3831] [Q&A] [answered]
Hi I was wondering about making a map which very likely will end up having lots of sub maps and tiled seems to not be super great to do that easily. Meanwhile, ldtk has perfect support to do this kind of worldstyle. 

Are there any plans to add support for both formats and/or are there people having experience to convert from ldtk maps to the workadventure formats? I am aware of the tiled export in ldtk, but it does the xml format rather than the json one, so I am not sure if that helps me at all.

**Comment by moufmouf** [comment]
Hey!

I'm not aware of anyone trying to use ldtk.

One thing you can try is design a map in ldtk, export it to tmx, load it in Tiled and export it back to TMJ to see if it helps.

Also, to make things easier, Tiled has a command line interface and it seems it can be used to import / export files. So if it works, you can automate the conversion from TMX to TMJ/JSON using the Tiled command line.

I'd be interested to know if you manage to have this working.

**Reply by MTRNord** [reply]
Yeah I will be looking into it :) one hurdle I already found is that areas in the export only have the start title while the size is in the json metadata encoded as a property instead. So it seems like there is a need for at least some post processing :)

**Reply by MTRNord** [reply]
Just for people reading along https://github.com/deepnight/ldtk/issues/1093 seems to be a major blocker for this right now. The rest seems to not really be much of an issue so far

**Reply by MTRNord** [reply, marked as answer]
I started to work on a post/pre processing script in https://gist.github.com/MTRNord/3567d833d06586039b94fe93bcde82f3 which fixes up some of the bugs I encounter by fetching the data from the ldtk json :)

**Reply by MTRNord** [reply]
I also just realised that ldtk has no concept of layer properties except for entities which are object layers in tiled which means there needs to be some manual helpers there to manage those externally sadly. But not a huge deal.
